### ISSUE TITLE
back-project

### ISSUE THREAD

**Issue by XuYunqiu** [NVlabs/nvdiffrast#137]
Hi, thanks for the great work.
I'm wondering is there any way to gather pixels on 2D image and back-project them onto 3D vertex/faces (like inverse render pipeline)?

**Comment by XuYunqiu** [comment]
Thanks for the prompt reply and sorry for the ambiguity. I'm not familiar enough with some details of the render process.
In my case, I have the index of some pixels on the 2D image plane, and I want to get the corresponding 3D faces.

The `rast[:, x, y, 3]` is the corresponding face (i.e., triangle_id) of the pixel (x, y) in the rendered image,
where `rast, rast_db = peeler.rasterize_next_layer()`.
Is that correct?

**Comment by s-laine** [comment]
That's correct, except you have _x_ and _y_ coordinates flipped. Also note that the value is offset by one, so you need to subtract 1 from the value in `rast[:, y, x, 3]` to get the triangle ID. A value of 0 means there was no triangle in that pixel.

**Comment by XuYunqiu** [comment]
got it, thanks again for your kind reminder
